src/c++/library/grpc_client.cc

```diff
@@ -91,51 +89,57 @@ GetChannelStub(
   static const size_t max_share_count =
       std::stoul(GetEnvironmentVariableOrDefault(
           "TRITON_CLIENT_GRPC_CHANNEL_MAX_SHARE_COUNT", "6"));
-  static std::atomic<int> channel_count{0};
-  auto current_idx = channel_count.fetch_add(1);
-  const auto& channel_itr = grpc_channel_stub_map_.find(
-      url + std::to_string(current_idx / max_share_count));
+  const auto& channel_itr = grpc_channel_stub_map_.find(url);
   if (channel_itr != grpc_channel_stub_map_.end()) {
-    return channel_itr->second;
-  } else {
-    grpc::ChannelArguments arguments;
-    arguments.SetMaxSendMessageSize(MAX_GRPC_MESSAGE_SIZE);
-    arguments.SetMaxReceiveMessageSize(MAX_GRPC_MESSAGE_SIZE);
-    // GRPC KeepAlive: https://github.com/grpc/grpc/blob/master/doc/keepalive.md
-    arguments.SetInt(
-        GRPC_ARG_KEEPALIVE_TIME_MS, keepalive_options.keepalive_time_ms);
-    arguments.SetInt(
-        GRPC_ARG_KEEPALIVE_TIMEOUT_MS, keepalive_options.keepalive_timeout_ms);
-    arguments.SetInt(
-        GRPC_ARG_KEEPALIVE_PERMIT_WITHOUT_CALLS,
-        keepalive_options.keepalive_permit_without_calls);
-    arguments.SetInt(
-        GRPC_ARG_HTTP2_MAX_PINGS_WITHOUT_DATA,
-        keepalive_options.http2_max_pings_without_data);
-    arguments.SetInt("client_channel_idx", current_idx);
```

Comment:
Why was this added and why has this argument been removed?

> channels must have different channel args to prevent re-use so define a use-specific channel arg such as channel number

Source: https://grpc.io/docs/guides/performance/

Does 1:1 mapping between the stubs and channel ensure that this case do not occur? Is it safe to just keep this client argument?

Reply:
That is probably why it was added. That slipped out of my mind and thus deleted, will add it back